src/test/spec/unified_runner/operation.rs

```diff
@@ -1919,6 +1938,195 @@ impl TestOperation for RenameCollection {
     }
 }
 
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub(super) struct Loop {
+    operations: Vec<Operation>,
+    store_errors_as_entity: Option<String>,
+    store_failures_as_entity: Option<String>,
+    store_successes_as_entity: Option<String>,
+    store_iterations_as_entity: Option<String>,
+}
+
+impl TestOperation for Loop {
+    fn execute_test_runner_operation<'a>(
+        &'a self,
+        test_runner: &'a mut TestRunner,
+    ) -> BoxFuture<'a, ()> {
+        async move {
+            if let Some(id) = &self.store_errors_as_entity {
+                let errors = Bson::Array(vec![]);
+                test_runner.insert_entity(id, errors.into());
+            }
+            if let Some(id) = &self.store_failures_as_entity {
+                let failures = Bson::Array(vec![]);
+                test_runner.insert_entity(id, failures.into());
+            }
+            if let Some(id) = &self.store_successes_as_entity {
+                let successes = Bson::Int64(0);
+                test_runner.insert_entity(id, successes.into());
+            }
+            if let Some(id) = &self.store_iterations_as_entity {
+                let iterations = Bson::Int64(0);
+                test_runner.insert_entity(id, iterations.into());
+            }
+
+            let continue_looping = Arc::new(AtomicBool::new(true));
+            let continue_looping_handle = continue_looping.clone();
+            ctrlc::set_handler(move || continue_looping_handle.store(false, Ordering::SeqCst))
+                .expect("Failed to set ctrl-c handler");
+
+            while continue_looping.load(Ordering::SeqCst) {
+                for operation in &self.operations {
+                    let result = match operation.object {
+                        OperationObject::TestRunner => {
+                            panic!("Operations within a loop must be entity operations")
+                        }
+                        OperationObject::Entity(ref id) => {
+                            operation.execute_entity_operation(id, test_runner).await
+                        }
+                    };
+                    match result {
```

Comment:
since we have a lot of nested matches and if lets, this might be more readable as one larger, flat match statement,
e.g.
```rust
match (result, operation.expectation) {
    // Have an entity result, expected one, need to check if equal
    (Ok(Entity::Bson(entity)), Expectation::Result { Some(expected_value), save_as_entity }) => { ... },

    // Got an ok response and expected one, but don't care about what it is, always success
    (Ok(_), Expectation::Result { None, _ }) => { ... }

    // Got an ok result, but expected an error, always a failure
    (Ok(entity), Expectation::Error(expected_error) => { ... }

    // ignore, always success
    (Ok(_), Expectation::Ignore) => { ... }

    // Got an error, but expected a result
    (Err(e), Expectation::Result { ... } | Expectation::Ignore) => { ... }

    // Got an error, expected an error
    (Err(e), Expectation::Error(expected_error)) => { ... }
}
```
To me this is a bit easier to reason about and also has the benefit of having the compiler verify that we've handled all cases properly. Also could combine the first two cases and keep a nested `if let`. Anyways, this is just my preference, feel free to leave it as-is if you think the current version is more readable.